---
title: "Control_pipeline_DGEseqandGSEAprep"
output: html_notebook
---

This is an [R Markdown](http://rmarkdown.rstudio.com) Notebook. When you execute code within the notebook, the results appear beneath the code. 

Try executing this chunk by clicking the *Run* button within the chunk or by placing your cursor inside it and pressing *Cmd+Shift+Enter*. 

```{r}
####  DESeq2 Script for Differential Gene Expression Analysis in 
      # Functional Genomics BIOL: 6850
### Resources and Citations:
# STANDARD PIPELINE
#Love et al 2016 DESeq2 GenomeBiology
# https://bioconductor.riken.jp/packages/2.14/bioc/vignettes/DESeq2/inst/doc/DESeq2.pdf
# http://master.bioconductor.org/packages/release/workflows/vignettes/rnaseqGene/inst/doc/rnaseqGene.html

## Load the DESeq2 library 
library(DESeq2)

##########   1.3 Input data   ##############

### Input the count data, the gene(/transcript) count matrix and labels
countdata <- as.matrix(read.csv("STND_PIPELINE_gene_count_matrix.csv", row.names="gene_id"))
dim(countdata)
head(countdata)

### Input the meta data or phenotype data
coldata <-(read.csv("LIZARD_pheno_meta.csv", header=TRUE, row.names=1))
dim(coldata)
head(coldata)

#Check all sample IDs in colData are also in CountData and match their orders
all(rownames(coldata) %in% colnames(countdata))
countdata <- countdata[, rownames(coldata)]
all(rownames(coldata) == colnames(countdata))

## Create the DESEQ dataset and define the statistical model (page 6 of the manual)
dds <- DESeqDataSetFromMatrix(countData = countdata, colData=coldata,  design = ~treatment)
dds

#####   Prefiltering    Manual - starting at  1.3.6 
dds <- dds[ rowSums(counts(dds)) > 20, ]
dds
dim(dds)

## set factors for statistical analyses
dds$condition <- factor(dds$treatment, levels=c("control","heat_shock"))

######     1.4 Differential expression analysis
dds <- DESeq(dds)
res <- results(dds)
res

resOrdered <- res[order(res$padj),]
resOrdered
summary(res)
sum(res$padj < 0.1, na.rm=TRUE)

res05 <- results(dds, alpha=0.05)
summary(res05)
sum(res05$padj < 0.05, na.rm=TRUE)

###    1.5.1 MA-plot
plotMA(res, main="DESeq2", ylim=c(-8,8))

##  1.5.2 Plot counts - sanity check!
## plotCounts(dds, gene="FUN_025750", intgroup="treatment")
## plotCounts(dds, gene=which.min(res$padj), intgroup="treatment")
## dds
  
##  Write your results to a file 
write.csv(as.data.frame(resOrdered), file="DGESeq_results.csv")  
```


```{r}
## 2.1.2 Extracting transformed values
rld <- rlog(dds)
vsd <- varianceStabilizingTransformation(dds)
head(assay(rld), 3)
  
library("pheatmap")
topVarGenes <- head(order(rowVars(assay(vsd)), decreasing = TRUE), 50)
mat  <- assay(vsd)[ topVarGenes, ]
mat  <- mat - rowMeans(mat)
anno <- as.data.frame(colData(vsd)[, "treatment", drop=FALSE])
pheatmap(mat, annotation_col = anno)
  
# 2.2.2 Heatmap of the sample-to-sample distances
sampleDists <- dist(t(assay(rld)))
library("RColorBrewer")
sampleDistMatrix <- as.matrix(sampleDists)
rownames(sampleDistMatrix) <- paste(rld$treatment)
colors <- colorRampPalette( rev(brewer.pal(9, "Blues")) )(255)
pheatmap(mat, annotation_col = anno)
  
# 2.2.3 Principal component plot of the samples
plotPCA(rld, intgroup=c("treatment"))

```

```{r}
############ Preparing Data for GSEA and Cytoscape.  #############
  
### Merge 'gene names' with DGE results by Gene Model
  
## Import Annotation file with results from Blast to databases
Anno <- read.csv("SceUnd1.0_top24_CompliedAnnotation.csv", stringsAsFactors = FALSE, na.strings=c("", "NA"))
  
## Import the DGE results file make sure the gene model name is 'gene_id' to match annotation file
## For GSEA- use HGCNA gene ID symbols. Have to merge ANNO and DGESeq_results by gene ID. Must merge on ID type used in gene matrix file. (FUN ID’s here). Rename ID column in DGEresults to be the same as annotation file with the same ID type (FUN ID here). Then this will merge annotation data and DGEresults .csv. It will merge all columns and be in a temp dataframe called DGE_Anno_Rank. Next you will only save the HGCNA containing row (look at name in file)(GeneID here) and the “rank” column. 
DGEresults <- read.csv("DGESeq_results.csv", stringsAsFactors = FALSE)

## Rename first column so it matches "gene_id" in annotation file
names(DGEresults)[1]<- "gene_id" 

#Merge anno with DGE results
DGE_Anno <- merge(Anno,DGEresults,by="gene_id",all.y = FALSE)

```

```{r}
View(DGE_Anno)
```


```{r}
############################# Make ranked list for GSEA ####################
DGE_Anno_Rank <-  within(DGE_Anno, rank <- sign(log2FoldChange) * -log10(pvalue))
DGErank = subset(DGE_Anno_Rank, select = c(GeneID, rank) )
DGErank_withName <- na.omit(DGErank)

##write.table(as.data.frame(DGErank_withName), file="DGErank_withname2.txt", row.names=FALSE)
write.table(DGErank_withName, file="DGErank_withname_forGSEA.rnk", sep="\t", row.names=FALSE, quote=FALSE)

####  We also need the normalized expression DATA
nt <- normTransform(dds) # defaults to log2(x+1)
NormTransExp<-assay(nt)
gene <-gsub("^[^-]+-", "", rownames(NormTransExp))
NormTransExpIDs  <-cbind(gene,NormTransExp)

write.table(as.data.frame(NormTransExpIDs), file="NormTransExpIDs.txt", row.names=FALSE)
```


```{r}
View(DGErank_withName)
```





```{r}
