## Standard Pipeline
This directory contains our standard pipeline script, our DE gene expression (.R) script, and phenotypic data necessary for running the R script.  To complete this analysis, you'll need to edit the pipeline script to reflect the directories you want. You should include your userID to ensure this happens in *your* directories. You'll also need to download the correct [reference genome assembly](https://s3.ap-northeast-1.wasabisys.com/gigadb-datasets/live/pub/10.5524/100001_101000/100913/SceUnd1.0_top24.fasta.gz) and [annotation file](https://s3.ap-northeast-1.wasabisys.com/gigadb-datasets/live/pub/10.5524/100001_101000/100913/SceUnd1.0_top24.gff3) and place these into a reference directory. Be sure to look at the specified resources at the top of the script. This pipeline works best when you `gunzip` the reference.

For the DE analysis in R, you will also need to download the [compiled annotation](https://s3.ap-northeast-1.wasabisys.com/gigadb-datasets/live/pub/10.5524/100001_101000/100913/SceUnd1.0_top24_CompliedAnnotation.csv).

RNA-seq scripts in this repository are adapted from those in the class [github from Functional Genomics 2024](https://github.com/Schwartz-Lab-at-Auburn/FunGen2024).
